### PR TITLE
[git] address git commit textarea and git clone bugs

### LINKED ISSUE #3543: [git] git clone should not be enabled when no workspace is present
Currently, if a user attempts to run the command `Git: Clone...` without a workspace, Theia does not do anything and does not provide any feedback. Instead of being silent I propose we either disable the command or display a feedback message.

### LINKED ISSUE #3544: [git] commit textarea resizes oddly
When clicking the `Commit` textarea in the `Git` view widget, some weird rendering occurs due to borders.

git-commit-sad

### PR DESCRIPTION
**Pull request by vince-fugnitto**
Fixes #3544
Fixes #3543

Signed-off-by: Vincent Fugnitto <vincent.fugnitto@ericsson.com>

### CODE REVIEW

File: packages/git/src/browser/git-view-contribution.ts [hunk 1]
```diff
@@ -261,6 +263,7 @@ export class GitViewContribution extends AbstractViewContribution<GitWidget>
             isVisible: () => this.syncService.canPublish()
         });
         registry.registerCommand(GIT_COMMANDS.CLONE, {
+            isEnabled: () => !!this.workspaceService.tryGetRoots().length,
```
Comment (elaihau) [inline packages/git/src/browser/git-view-contribution.ts]: would `this.workspaceService.opened` work?

Comment (vince-fugnitto) [reply packages/git/src/browser/git-view-contribution.ts]: yes that's shorter :)

**Comment by westbury** [comment]
Note that we also have PR #3438.  That PR has the purpose of restructuring the Git view to support other Source Control Managers.  However a side-effect of that PR is that it also fixes #3543.  Not only may there not be a workspace, but the selected project may not be a Git repository (though it is hard for a user to get into this situation using a standard Theia distribution).  PR #3438 adds an additional widget that is shown in the view when there is no selected SCM repository or where the repository is not of a type that is supported by any extensions.

**Comment by vince-fugnitto** [comment]
> Note that we also have PR #3438. That PR has the purpose of restructuring the Git view to support other Source Control Managers. However a side-effect of that PR is that it also fixes #3543. Not only may there not be a workspace, but the selected project may not be a Git repository (though it is hard for a user to get into this situation using a standard Theia distribution). PR #3438 adds an additional widget that is shown in the view when there is no selected SCM repository or where the repository is not of a type that is supported by any extensions.

Thanks for letting me know, can you rebase your PR? There seems to be some conflicts. The only way that I am fixing #3543 is by disabling the command when no workspace is present, much like other commands which do not make sense in the context of no workspace.

**Comment by vince-fugnitto** [comment]
@jbicker do you know how we can get consistent styling for `textarea` across browsers, I noticed my patch works correctly for chrome, but firefox the issue persists.

**Comment by vince-fugnitto** [comment]
closing for the moment until a solution is made for consistent styling of `textarea` across different browsers

**Pull request closed without merging**